lib/shelley/test/integration/Main.hs

```diff
@@ -276,34 +285,27 @@ specWithServer testDir (tr, tracers) = aroundAll withContext
                 let event = PoolGarbageCollectionEvent epochNo certificates
                 liftIO $ do
                     traceWith tr $ MsgPoolGarbageCollectionEvent event
-                    atomicModifyIORef' eventsRef ((, ()) . (event :))
+                    atomicModifyIORef' eventsRef ((,()) . (event :))
                 pure certificates
 
     withServer dbDecorator onReady = bracketTracer' tr "withServer" $
         withSMASH testDir $ \smashUrl -> do
-            minSev <- nodeMinSeverityFromEnv
-            testPoolConfigs <- poolConfigsFromEnv
-            extraLogDir <- (fmap (,Info)) <$> testLogDirFromEnv
-            withCluster
-                tr'
-                minSev
-                testPoolConfigs
-                testDir
-                extraLogDir
-                (onClusterStart (onReady $ T.pack smashUrl) dbDecorator)
+            clusterCfg <- localClusterConfigFromEnv era
+            withCluster tr' testDir clusterCfg $
+                onClusterStart (onReady $ T.pack smashUrl) dbDecorator
 
     tr' = contramap MsgCluster tr
-    setupFaucet = do
+    setupFaucet conn = do
         traceWith tr MsgSettingUpFaucet
         let rewards = (,Coin $ fromIntegral oneMillionAda) <$>
                 concatMap genRewardAccounts mirMnemonics
-        moveInstantaneousRewardsTo tr' testDir rewards
+        moveInstantaneousRewardsTo tr' conn testDir rewards
```

Comment:
Btw, I had been thinking one cold probably factor out `setupFaucet` from integration:main and local-cluster to a simpler variant of withCluster (or even the same).

Reply:
True - probably some other things could be factored out too.